sparse/core.py

```diff
@@ -1323,6 +1323,60 @@ def stack(arrays, axis=0):
                sorted=(axis == 0) and all(a.sorted for a in arrays))
 
 
+def triu(x, k=0):
+    """
+    Calculates the equivalent of np.triu(x, k) for COO.
+
+    Parameters
+    ----------
+    x : COO
+        The input array.
+    k : int
+        The diagonal below which elements are set to zero.
+
+    Returns
+    -------
+    COO
+        The output upper-triangular matrix.
+    """
+    if not x.ndim >= 2:
+        raise NotImplementedError('sparse.triu is not implemented for scalars or 1-D arrays.')
+
+    mask = x.coords[-2] + k <= x.coords[-1]
```

Comment:
Interesting. Was not aware of this behavior in NumPy. Raised issue ( https://github.com/numpy/numpy/issues/10310 ) to see if we can get it documented in NumPy.